### PR TITLE
refactor: persist forkchoice state

### LINKED ISSUE #1116: [Refactor] Persist the fork choice store.
The motivation behind this is to be able to react to new blocks in fork choice by

1. Reading the store.
2. Applying state transition.
3. Saving the store.

### LINKED ISSUE #1117: [Refactor Fork choice] Measure how long it takes to interact with the store db
We need to see how long it takes to persist and how long it takes to read the store. We should be able to see this in graphana.

### PR DESCRIPTION
**Pull request by avilagaston9**
**Motivation**

In order to reduce complexity, we need to reduce concurrency as much as possible.

**Description**

This PR removes the use of the `GenServer` state, storing it into the db, taking the first step towards removing the `GenServer` calls.

Additionally, it adds some metrics on store persisting and fetching:
image

Closes #1116 
Closes #1117

### CODE REVIEW

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex [hunk 1]
```diff
@@ -93,6 +95,7 @@ defmodule LambdaEthereumConsensus.ForkChoice do
         prune_old_states(last_finalized_checkpoint.epoch, new_finalized_checkpoint.epoch)
 
         GenServer.cast(from, {:block_processed, block_root, true})
+        persist_fork_choice_store(new_store)
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:98]: I would persist before calling the cast. As a general rule it's useful that the state is stored and consistent before others are made aware of it. As a matter of fact, if we persist it before launching the recompute_head async task, we can remove the persist call in recompute-head.

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex [hunk 2]
```diff
@@ -219,4 +228,19 @@ defmodule LambdaEthereumConsensus.ForkChoice do
       Logger.debug("[Fork choice] Store persisted")
     end)
   end
+
+  defp persist_fork_choice_store(store) do
+    :telemetry.span([:fork_choice, :persist], %{}, fn ->
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:233]: Let's have the telemetry call on the db methods directly, so they are called no matter who is storing/fetching the store.

**Review by Arkenan — APPROVED** [review]
LGTM with some small comments.

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex (hunk 2 above)
Comment (avilagaston9) [reply lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:233]: Done!

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex (hunk 1 above)
Comment (avilagaston9) [reply lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:98]: That's true. Done!

**Pull request merged**